src/platform/tls_openssl.c

```diff
@@ -1260,6 +1260,118 @@ CxPlatTlsParamSet(
     return QUIC_STATUS_NOT_SUPPORTED;
 }
 
+//
+// Mappings taken from the following .NET definitions.
+// https://github.com/dotnet/runtime/blob/69425a7e6198ff78131ad64f1aa3fc28202bfde8/src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Linux.cs
+// https://github.com/dotnet/runtime/blob/69425a7e6198ff78131ad64f1aa3fc28202bfde8/src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c
+// https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Net.Security/src/System/Net/Security/TlsCipherSuiteData.Lookup.cs
+//
+
+#define TLS_AES_128_GCM_SHA256          0x1301
+#define TLS_AES_256_GCM_SHA384          0x1302
+#define TLS_CHACHA20_POLY1305_SHA256    0x1303
+
+static
+QUIC_STATUS
+CxPlatMapCipherSuite(
+    _Inout_ QUIC_HANDSHAKE_INFO* HandshakeInfo
+    )
+{
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+
+    HandshakeInfo->KeyExchangeAlgorithm = 0;
+    HandshakeInfo->KeyExchangeStrength = 0;
+
+    switch (HandshakeInfo->CipherSuite) {
+        case TLS_AES_128_GCM_SHA256:
+            HandshakeInfo->CipherAlgorithm = 0x660E;
+            HandshakeInfo->CipherStrength = 128;
+            HandshakeInfo->Hash = 0x800c;
+            HandshakeInfo->HashStrength = 0;
+            break;
+        case TLS_AES_256_GCM_SHA384:
+            HandshakeInfo->CipherAlgorithm = 0x6610;
+            HandshakeInfo->CipherStrength = 256;
+            HandshakeInfo->Hash = 0x800d;
+            HandshakeInfo->HashStrength = 0;
+            break;
+        case TLS_CHACHA20_POLY1305_SHA256:
+            HandshakeInfo->CipherAlgorithm = 0;
+            HandshakeInfo->CipherStrength = 256;
+            HandshakeInfo->Hash = 0x800c;
+            HandshakeInfo->HashStrength = 0;
+            break;
+        default:
+            Status = QUIC_STATUS_NOT_SUPPORTED;
+            break;
+    }
+
+    return Status;
+}
+
+static
+uint32_t
+CxPlatMapVersion(
+    _In_ const char* Version,
+    _In_ BOOLEAN IsServer
+    )
+{
+    //
+    // protocolVersion points to a static ASCII string that's one of:
+    //     TLSv1
+    //     TLSv1.1
+    //     TLSv1.2
+    //     TLSv1.3
```

Comment:
Anything less than TLSv1.3 is unsupported and I'd prefer not to have code that allows for it at all. We should either return an error or some "unsupported" version value.